dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiChatCompletionClient.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Represents a client for interacting with the chat completion gemini models by Vertex AI.
+/// </summary>
+// todo: remove this class when gemini vertex ai (preview api) support non-streaming
+internal sealed class VertexAIGeminiChatCompletionClient : GeminiChatCompletionClient
```

Comment:
Are we going to have other specializations of GeminiChatCompletionClient ? 
Is VertexAIGemini completely different from other model supported by Google?

Reply:
VertexAI should have same implementation like GoogleAI only other endpoints and requests. But current version of VertexAI rest api (preview) has limitations, so this is only reason why classes VertexAIGeminiTextGenerationClient and VertexAIGeminiChatCompletionClient exists. I intentionally added TODO comments to these classes. I hope before I finish connector, vertex ai will be stable and I will able to remove this classes completely.
I can delete VertexAIGeminiTextGenerationClient but VertexAIGeminiChatCompletionClient have to exist, at least for now.